File: 1_two_sum/two_sum.py
from typing import List, Dict


class Solution:
    def two_sum(self, nums: List[int], target: int) -> List[int]:
        nmap: Dict[int, int] = {}
        for i, num in enumerate(nums):
            complement = target - num
            if complement in nmap:
                return [i, nmap[complement]]
            nmap[num] = i
        return []

if __name__ == '__main__':
    solution = Solution()

    nums = [2,7,11,15]
    target = 9

    result = solution.two_sum(nums, target)
    print(result)

File: 1_two_sum/two_sum.go
package main

import "fmt"

type Solution struct {
}

func (s *Solution) two_sum(nums []int, target int) []int {
	nmap := make(map[int]int)
	for i, num := range nums {
		complement := target - num
		if j, found := nmap[complement]; found {
			return []int{i, j}
		}
		nmap[num] = i
	}
	return []int{}

}

func main() {
	solution := new(Solution)
	nums := []int{2, 7, 11, 15}
	target := 9
	result := solution.two_sum(nums, target)
	fmt.Println(result)
}
